bsip-0073.md

```diff
@@ -0,0 +1,210 @@
+
+If a margin call order appears, either due to feed price changed, or due to
+collateral or debt changed (which implies feed price is valid),
+* current logic is to only match it with the limit orders on the opposite;
+* after the hard fork, the new logic would be:
+  * firstly match it with the limit orders on the opposite whose buy prices
+    are higher than feed price;
+  * secondly match it with the settle orders with `fill_asap` set to `true`,
+    the matching price would be feed price, in other words, treat the settle
+    orders as maker limit orders buying at feed price;
+  * lastly match it with remaining limit orders on the opposite.
+
+### When feed price changed
+
+When feed price changed, either due to a new price is published, or due
+to an old price expired, or due to asset options changed, if the new feed
+price is valid, and if the new feed price in the direction of "X debt asset
+per collateral asset" is higher than the old feed price or the old feed
+price was invalid,
+* current logic doesn't handle settle orders,
+* after the hard fork, the new logic would be:
+  * if there are limit orders selling collateral asset below the new feed
+    price, and there are settle orders whose `fill_asap` is `true`,
+    match the settle orders with the limit orders, in other words, treat
+    those settle orders as taker limit orders buying at the new feed price.
+
+### Do not update `force_settled_volume` before the delay
+
+When a settle order is matched and (partially) filled before the delay, 
+it doesn't affect `force_settled_volume` (which indicates how much debt
+asset has been force-settled in current maintenance interval).
+
+## Processing settle orders after the delay
+
+Currently, when filling a settle order after the delay (note: it implies
+some conditions E.G. price feed is valid and total settled volume in current
+maitenance interval doesn't exceed maximum allowed volume),
+the settle order will be matched against the debt position with the least
+collateral ratio, the fill price in the direction of "X debt asset
+per collateral asset" would be
+`fill_price = feed_price * (1 + foce_settlement_offset)`.
+
+After the hard fork, when processing a settle order after the delay,
+* firstly try to match it with the margin calls and limit orders selling
+  below `fill_price`, in other words, treat it as a taker limit order buying
+  at `fill_price`.
+  Note: this step doesn't affect `force_settled_volume`.
+* if the settle order still exists, process it with the logic before the
+  hard fork.
+  Note: this step does affect `force_settled_volume` as before.
+
+## API
+
+APIs which return `settle_order_object` need to return the new `fill_asap`
+field.
+
+APIs which return combined order book can combine settle orders whose
+`fill_asap` is true with limit orders in the same direction.
+
+## CLI
+
+There was a `settle_asset` command in CLI.
+
+Need to add a new command E.G. `settle_asset_ext` in CLI so that users would
```

Comment:
How about adding an optional boolean flag to the parameter list of `settle_asset`?

Reply:
Need to test whether this approach would break existing client applications which relies on wallet API. I know node API supports optional parameters already, but wallet API is a bit different than node API.